elements/power.lua

```diff
@@ -160,31 +167,41 @@ local function UpdateColor(self, event, unit)
 		r, g, b = self:ColorGradient((element.cur or 1) + adjust, (element.max or 1) + adjust, unpack(element.smoothGradient or self.colors.smooth))
 	end
 
-	if(color) then
-		r, g, b = color[1], color[2], color[3]
-	end
+	if(atlas) then
+		element:SetStatusBarTexture(atlas)
+		element:SetStatusBarColor(1, 1, 1)
+	else
+		if(color) then
+			r, g, b = color[1], color[2], color[3]
+		end
 
-	if(b) then
-		element:SetStatusBarColor(r, g, b)
+		if(b) then
+			if(element.__texture) then
+				element:SetStatusBarTexture(element.__texture)
+			end
 
-		local bg = element.bg
-		if(bg) then
-			local mu = bg.multiplier or 1
-			bg:SetVertexColor(r * mu, g * mu, b * mu)
+			element:SetStatusBarColor(r, g, b)
+
+			local bg = element.bg
+			if(bg) then
+				local mu = bg.multiplier or 1
+				bg:SetVertexColor(r * mu, g * mu, b * mu)
+			end
 		end
 	end
 
 	--[[ Callback: Power:PostUpdateColor(unit, r, g, b)
 	Called after the element color has been updated.
 
-	* self - the Power element
-	* unit - the unit for which the update has been triggered (string)
-	* r    - the red component of the used color (number)[0-1]
-	* g    - the green component of the used color (number)[0-1]
-	* b    - the blue component of the used color (number)[0-1]
+	* self  - the Power element
+	* unit  - the unit for which the update has been triggered (string)
+	* r     - the red component of the used color (number?)[0-1]
+	* g     - the green component of the used color (number?)[0-1]
+	* b     - the blue component of the used color (number?)[0-1]
+	* atlas - the atlas used instead of color (string)
```

Comment:
I reckon it should be `string?` just like `number?` since it's not guaranteed to be there.